src/database/Database.cpp

```diff
@@ -366,6 +366,23 @@ Database::initialize()
     // normally you do not want to touch this section as
     // schema updates are done in applySchemaUpgrade
 
+    if (isSqlite())
+    {
+        mSession << "PRAGMA journal_mode = OFF";
+        mSession << "PRAGMA page_size = 65536";
+        mSession << "VACUUM";
+        if (mApp.getConfig().DATABASE.value == "sqlite3://:memory:")
+        {
+            // Sqlite will ignore the 'WAL' mode setting if it's in-memory;
```

Comment:
does that mean that line 141 in `doSqliteSpecificOperation` is wrong?

Reply:
It means line 141 has no effect in in-memory mode. The thing is that in-memory mode can have journal_mode as either MEMORY or OFF, and since we turn it OFF in this patch we need to turn it back on to _something_ valid, WAL is not an option, it's silently ignored.

Or, as you say, we can just predicate this whole block on not-being-in-memory-mode, which I think I'll do.